src/redis-cli.c

```diff
@@ -2172,9 +2179,11 @@ static int noninteractive(int argc, char **argv) {
         argv = zrealloc(argv, (argc+1)*sizeof(char*));
         argv[argc] = readArgFromStdin();
         retval = issueCommand(argc+1, argv);
+        sdsfree(argv[argc]);
     } else {
         retval = issueCommand(argc, argv);
     }
+    freeConvertedSds(argc, argv);
```

Comment:
i think this one should move to the caller, it's confusing to see it here (you don't know how these arguments were created).

Reply:
We can't move it to the caller since there is a zrealloc in `noninteractive`.

Reply:
so how about moving the call to `convertToSds` to be inside `noninteractive`?

even then, this `realloc` still makes it a bit odd since it means `noninteractive` assumes it knows what `convertToSds` does, and that this pointer can be realloced, but at least all of it is inside that function, and it doesn't assume anything about how the caller uses it.

Reply:
Fixed that.